@jonringer build fails, attaching log

`builder for '/nix/store/h93lhzqh9i02jl85iijqrivv0wd317qz-telegram-desktop-2.7.5.drv' failed with exit code 1; last 10 log lines:
  [457/912] Building CXX object Telegram/CMakeFiles/Telegram.dir/SourceFiles/history/view/controls/history_view_voice_record_bar.cpp.o
  FAILED: Telegram/CMakeFiles/Telegram.dir/SourceFiles/history/view/controls/history_view_voice_record_bar.cpp.o 
  /nix/store/35pnk5kwi26m3ph2bc7dxwjnavpzl8cn-gcc-wrapper-10.3.0/bin/g++ -DDESKTOP_APP_DISABLE_CRASH_REPORTS -DDESKTOP_APP_USE_PACKAGED -DDESKTOP_APP_USE_PACKAGED_FONTS -DG_LOG_DOMAIN=\"Telegram\" -DLOTTIE_USE_CACHE -DLOT_BUILD -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_DEBUG -DQT_NO_KEYWORDS -DQT_WAYLANDCLIENT_LIB -DQT_WIDGETS_LIB -DQT_XKBCOMMON_SUPPORT_LIB -DTDESKTOP_API_HASH=d524b414d21f4d37f08684c1df41ac9c -DTDESKTOP_API_ID=611335 -DTDESKTOP_DISABLE_AUTOUPDATE -DTDESKTOP_LAUNCHER_BASENAME=telegramdesktop -ITelegram/Telegram_autogen/include -I../Telegram/SourceFiles -I../Telegram/lib_spellcheck -I../Telegram/ThirdParty/tgcalls -I../Telegram/ThirdParty/libtgvoip -I../Telegram/lib_base -I../Telegram/lib_rpl -I../Telegram/lib_crl -ITelegram/gen -I../Telegram/lib_tl -ITelegram/lib_ui/gen -I../Telegram/lib_ui -I../Telegram/lib_ui/emoji_suggestions -I../Telegram/lib_lottie -I../Telegram/lib_webrtc -I../Telegram/lib_storage -I../Telegram/lib_qr -I../Telegram/ThirdParty/QR/cpp -I../Telegram/lib_webview -I../Telegram/SourceFiles/payments -I../Telegram/ThirdParty/rlottie/inc -I../Telegram/ThirdParty/statusnotifieritem -isystem ../Telegram/ThirdParty/GSL/include -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtCore/5.15.2 -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtCore/5.15.2/QtCore -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtGui/5.15.2 -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtGui/5.15.2/QtGui -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtXkbCommonSupport/5.15.2 -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtXkbCommonSupport/5.15.2/QtXkbCommonSupport -isystem /nix/store/rawg7zhdyc0ca3mw5adrbja4kfcgn8vp-qtwayland-5.15.2-dev/include/QtWaylandClient/5.15.2 -isystem /nix/store/rawg7zhdyc0ca3mw5adrbja4kfcgn8vp-qtwayland-5.15.2-dev/include/QtWaylandClient/5.15.2/QtWaylandClient -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtCore -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/./mkspecs/linux-g++ -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtGui -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtWidgets -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtNetwork -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtDBus -isystem /nix/store/6ypchrs5rjvkrqlzm5xi8iss2s4akjgd-qtbase-5.15.2-dev/include/QtXkbCommonSupport -isystem /nix/store/rawg7zhdyc0ca3mw5adrbja4kfcgn8vp-qtwayland-5.15.2-dev/include/QtWaylandClient -isystem /nix/store/95jq4b7sxq5k718yfp6gf7lhs1k0ccf4-tl-expected-unstable-2019-11-11/share -isystem /nix/store/d9zs9xg86lhqjqni0v8h2ibdrjb57fn4-glib-2.68.2-dev/include/glib-2.0 -isystem /nix/store/dcb3cyba5wl6qimv6vwdpbi0kg0g1nlb-glib-2.68.2/lib/glib-2.0/include -isystem /nix/store/ml5gsps86fgznfqriq9sr3i6ayk0sk0i-glibmm-2.64.5-dev/include/giomm-2.4 -isystem /nix/store/1ihcnyi7va2vql6wm9zlk7ssimhx5gbb-glibmm-2.64.5/lib/giomm-2.4/include -isystem /nix/store/ml5gsps86fgznfqriq9sr3i6ayk0sk0i-glibmm-2.64.5-dev/include/glibmm-2.4 -isystem /nix/store/1ihcnyi7va2vql6wm9zlk7ssimhx5gbb-glibmm-2.64.5/lib/glibmm-2.4/include -isystem /nix/store/jj7g9fz29y1ygram087c7np441daihvg-libsigc++-2.10.6/include/sigc++-2.0 -isystem /nix/store/jj7g9fz29y1ygram087c7np441daihvg-libsigc++-2.10.6/lib/sigc++-2.0/include -isystem /nix/store/d9zs9xg86lhqjqni0v8h2ibdrjb57fn4-glib-2.68.2-dev/include/gio-unix-2.0 -isystem /nix/store/1b2ahzbjq1fqnm7riz1a9s0w3rsnaysy-harfbuzz-2.8.0-dev/include/harfbuzz -isystem /nix/store/vsk1qc1na4izgz461vxkvn655yvarfr7-gtk+3-3.24.27-dev/include/gtk-3.0 -isystem /nix/store/x26skdnbygrryb189ghkk00f419vy9ci-atk-2.36.0-dev/include/atk-1.0 -isystem /nix/store/gp87jysb40b919z8s7ixcilwdsiyl0rp-cairo-1.16.0-dev/include/cairo -isystem /nix/store/qxak159dp4qrmvsck3kjkin4bszbr6ab-gdk-pixbuf-2.42.6-dev/include/gdk-pixbuf-2.0 -isystem /nix/store/c52730cidby7p2qwwq8cf91anqrni6lg-pango-1.48.4-dev/include/pango-1.0 -isystem /nix/store/amryzlagqsnql3wdnhvl43czf9334cwz-openal-soft-1.19.1/include/AL -isystem /nix/store/a5v6ahqg3br6l3v0jrvh9khqmxlh395a-webkitgtk-2.32.1-dev/include/webkitgtk-4.0 -isystem /nix/store/15pc94wg989gib883ngn7r772qjjmnay-libsoup-2.72.0-dev/include/libsoup-2.4 -isystem /nix/store/69ina81ww7rldl4xdqarx2lvghpnwijl-minizip-1.2.11/include/minizip -isystem /nix/store/jxsh87jv7v7xrg05xblryx184pm5jirz-libdbusmenu-qt-0.9.3+16/include/dbusmenu-qt5 -isystem /nix/store/2wf5w1crpvzfggdkn8kjgxqg5c3kiikp-kwayland-5.81.0-dev/include/KF5/KWayland/Client -isystem /nix/store/2wf5w1crpvzfggdkn8kjgxqg5c3kiikp-kwayland-5.81.0-dev/include/KF5 -O3 -DNDEBUG -fPIC -fno-strict-aliasing -pipe -Wall -W -Wno-unused-variable -Wno-unused-parameter -Wno-unused-function -Wno-switch -Wno-comment -Wno-unused-but-set-variable -Wno-missing-field-initializers -Wno-sign-compare -Wno-attributes -Wno-parentheses -Wno-stringop-overflow -Wno-maybe-uninitialized -Wno-error=class-memaccess -Wno-register -std=gnu++17 -Winvalid-pch -include /build/tdesktop-2.7.5-full/build/Telegram/CMakeFiles/Telegram.dir/cmake_pch.hxx -MD -MT Telegram/CMakeFiles/Telegram.dir/SourceFiles/history/view/controls/history_view_voice_record_bar.cpp.o -MF Telegram/CMakeFiles/Telegram.dir/SourceFiles/history/view/controls/history_view_voice_record_bar.cpp.o.d -o Telegram/CMakeFiles/Telegram.dir/SourceFiles/history/view/controls/history_view_voice_record_bar.cpp.o -c ../Telegram/SourceFiles/history/view/controls/history_view_voice_record_bar.cpp
  {standard input}: Assembler messages:
  {standard input}:2776: Warning: end of file not at end of a line; newline inserted
  {standard input}:3172: Error: no such instruction: `_zn3rpl7details17consumer_han'
  g++: fatal error: Terminated signal terminated program cc1plus
  compilation terminated.
  [459/912] Building CXX object Telegram/CMakeFiles/Telegram.dir/SourceFiles/history/view/controls/history_view_compose_controls.cpp.o
  ninja: build stopped: subcommand failed.
cannot build derivation '/nix/store/0sjw4p2s2i4pwnryhf6mg9sllcc6q7nm-review-shell.drv': 1 dependencies couldn't be built
error: build of '/nix/store/0sjw4p2s2i4pwnryhf6mg9sllcc6q7nm-review-shell.drv' failed

Link to currently reviewing PR:                                                                               
https://github.com/NixOS/nixpkgs/pull/125538

1 package failed to build:
tdesktop

`

I guess the upstream source is broken then?

--- a/pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix
+++ b/pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix
@@ -2,11 +2,11 @@
 , pkg-config, cmake, ninja, python3, wrapGAppsHook, wrapQtAppsHook, removeReferencesTo
 , qtbase, qtimageformats, gtk3, libsForQt5, enchant2, lz4, xxHash
 , dee, ffmpeg, openalSoft, minizip, libopus, alsaLib, libpulseaudio, range-v3
-, tl-expected, hunspell, glibmm, webkitgtk
+, tl-expected, hunspell, glibmm, webkitgtk, libtgvoip
 # Transitive dependencies:
 , pcre, xorg, util-linux, libselinux, libsepol, epoxy
 , at-spi2-core, libXtst, libthai, libdatrie
-, xdg-utils
+, xdg-utils, libsysprof-capture, libpsl, brotli
 }:
 
 with lib;
@@ -59,10 +59,10 @@ in mkDerivation rec {
     qtbase qtimageformats gtk3 libsForQt5.kwayland libsForQt5.libdbusmenu enchant2 lz4 xxHash
     dee ffmpeg openalSoft minizip libopus alsaLib libpulseaudio range-v3
     tl-expected hunspell glibmm webkitgtk
-    tg_owt
+    tg_owt libtgvoip
     # Transitive dependencies:
     pcre xorg.libpthreadstubs xorg.libXdmcp util-linux libselinux libsepol epoxy
-    at-spi2-core libXtst libthai libdatrie
+    at-spi2-core libXtst libthai libdatrie libsysprof-capture libpsl brotli
   ];
 
   cmakeFlags = [

--- a/pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix
+++ b/pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix
@@ -20,27 +20,33 @@ with lib;
 
 let
   tg_owt = callPackage ./tg_owt.nix {};
-  webviewPatch = fetchpatch {
-    url = "https://raw.githubusercontent.com/archlinux/svntogit-community/013eff77a13b6c2629a04e07a4d09dbe60c8ca48/trunk/fix-webview-includes.patch";
-    sha256 = "0112zaysf3f02dd4bgqc5hwg66h1bfj8r4yjzb06sfi0pl9vl96l";
-  };
-
 in mkDerivation rec {
   pname = "telegram-desktop";
-  version = "2.7.4";
+  version = "2.7.5";
 
   # Telegram-Desktop with submodules
   src = fetchurl {
     url = "https://github.com/telegramdesktop/tdesktop/releases/download/v${version}/tdesktop-${version}-full.tar.gz";
-    sha256 = "1cigqvxa8lp79y7sp2w2izmmikxaxzrq9bh5ns3cy16z985nyllp";
+    sha256 = "sha256-9GxBw5ii9Musjq7D3KMf/P5BA4h690EgXRbhynHwO98=";
   };
+
+  patches = [
+    # fixes issue with ffmpeg>=4.4 crashes, hasn't been upstreamed yet
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/gentoo/gentoo/1c91884873968997be4b0c954169d04dc839f1db/net-im/telegram-desktop/files/tdesktop-2.7.4-voice-crash.patch";
+      sha256 = "sha256-inLXcP70yJlkkmdeXlc3HRL7Vt+Sf00LLJG33gwBKdY=";
+    })
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/gentoo/gentoo/1c91884873968997be4b0c954169d04dc839f1db/net-im/telegram-desktop/files/tdesktop-2.7.4-voice-ffmpeg44.patch";
+      sha256 = "sha256-p57LipNf7BDhVvNKRuicVqx0vU6IBL/Cvr5BAfLF4Hs=";
+    })
+  ];
 
   postPatch = ''
     substituteInPlace Telegram/lib_spellcheck/spellcheck/platform/linux/linux_enchant.cpp \
       --replace '"libenchant-2.so.2"' '"${enchant2}/lib/libenchant-2.so.2"'
     substituteInPlace Telegram/CMakeLists.txt \
       --replace '"''${TDESKTOP_LAUNCHER_BASENAME}.appdata.xml"' '"''${TDESKTOP_LAUNCHER_BASENAME}.metainfo.xml"'
-    patch -d Telegram/lib_webview -p1 < "${webviewPatch}"
   '';
 
   # We want to run wrapProgram manually (with additional parameters)

